platform/core/src/services/HangingProtocolService/HPMatcher.js

```diff
@@ -16,32 +19,32 @@ const match = (metadataInstance, rules, customAttributeRetrievalCallbacks) => {
     failed: [],
   };
 
+  const readValues = {};
+
   let requiredFailed = false;
   let score = 0;
 
   rules.forEach(rule => {
-    const attribute = rule.attribute;
+    const { attribute, source } = rule;
```

Comment:
what is `source` in the rule? we did not have this IIRC

Reply:
TODO - add documentation on the source attribute, or remove it.